app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,23 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path
+    end
+  end
+
+  def create
+    user = User.where(token: params[:token]).first
+    if user
+      user.password = params[:password]
```

Comment:
if password entered by user is wrong, or invalid. How are you going to handle that? You should use `if user && user.update(password: params[:password])`